gis/grass/Portfile

```
@@ -1,11 +1,300 @@
-# -*- coding: utf-8; mode: tcl; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=2:ts=2:sts=2
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
+PortGroup           wxWidgets 1.0
+PortGroup           github 1.0
+PortGroup           debug 1.0
 
+github.setup        OSGeo grass 8.0.0
 name                grass
-epoch               1
-version             6.4.6
-revision            5
-replaced_by         grass7
-PortGroup           obsolete 1.0
+set main_version    [join [lrange [split ${version} "."] 0 1] ""]
+revision            0
+set realVersion     ${version}
+distname            grass-${realVersion}
+maintainers         nomaintainer
+categories          gis
+license             GPL-2+
+platforms           darwin
 
+description         GRASS
+long_description    GRASS is a Geographic Information System (GIS) used for \
+                    geospatial data management and analysis.
+
+homepage            http://grass.osgeo.org/
+master_sites        ${homepage}grass[join [lrange [split ${realVersion} .] 0 1] {}]/source/
+
+checksums           rmd160  2192eb3ef991dd9645a7521f691aa9dc12bc4f16 \
+                    sha256  43231ae62874ace92b35d1457950749046aae374e4d9af9d9da22cf9c55d22a1 \
+                    size    64944539
+
+depends_build       port:pkgconfig
+depends_lib         port:bzip2 \
+                    port:freetype \
+                    port:fftw-3 \
+                    port:gdal \
+                    port:pdal \
+                    port:geos \
+                    port:tiff \
+                    port:libpng \
+                    port:liblas\
+                    path:lib/pkgconfig/cairo.pc:cairo \
+                    port:zlib \
+                    port:readline
+
+patchfiles          800-patch.diff
+
+configure.cflags    -DGL_SILENCE_DEPRECATION
+configure.ldflags   -Wl,-rpath,${prefix}/lib
+
+configure.args-append \
+    --prefix=${prefix}/share \
+    --exec-prefix=${prefix} \
```

MacPorts uses `configure.pre_args` for the prefix argument. If you're changing the prefix, that's the place to do it.

```suggestion
configure.pre_args \
      --prefix=${prefix}/share
configure.args-append \
    --exec-prefix=${prefix} \
```

${prefix}/share is a strange prefix. Why do this?

---

By default grass installs:

1. executable at `$prefix/bin/grass`
2. package/directory at `$prefix/grass80`

Setting `--prefix` and `--exec-prefix` installs these to more proper places (I now updated to install grass dir into $prefix/lib instead of $prefix/share, which seems more adequate).
I realise this is bit of a hack, I will look at if/how this can be fixed upstreams, but for the time being I'd suggest to leave this as it has been for grass7.

Content of grass directory:
```
grass80
├── AUTHORS
├── bin
│   └── ...
├── CHANGES
├── CITING
├── contributors_extra.csv
├── contributors.csv
├── COPYING
├── demolocation
│   └── ...
├── docs
│   └── ...
├── driver
│   └── ...
├── etc
│   └── ...
├── fonts
│   └── ...
├── GPL.TXT
├── gui
│   └── ...
├── include
│   └── ...
├── INSTALL
├── lib
│   └── ...
├── REQUIREMENTS.html
├── scripts
├── share
│   └── ...
├── translators.csv
└── utils
    └── ...
```